pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -102,7 +104,8 @@ type ensureAzResult int
 
 const (
 	ensureAzResultNotReady ensureAzResult = iota
-	ensureAzResultModified
+	ensureAzResultProjectClaimModified
+	ensureAzResultBothModified
```

Comment:
what's the reason we need to distinguish which resource got modified or if both got modified? My understanding of this return value is, when a resource was modified, return and let the next run of reconcile continue, independent of which resource was modified?